packages/core/src/components/SupportButton/SupportButton.tsx

```diff
@@ -124,8 +128,11 @@ export const SupportButton = ({ children }: PropsWithChildren<Props>) => {
               {child}
             </ListItem>
           ))}
-          {items &&
-            items.map((item, i) => <SupportListItem item={item} key={i} />)}
+          {supporters
```

Comment:
Is the intention to override the support items entirely from the `app-config.yaml`. I would think there might be a need to merge the 2 lists. One for the global / general support and another for plugin / instance specific usage. The trick being which gets priority and how should then be ordered in relation to each other?

Reply:
Hey @andrewthauer, thanks for the feedback, just one question, what do you exactly mean on your last comment? I don't know if I understood well, do you mean merge both lists, the original and mine (supportsList), and then decide how to display the merged list? e.g. if I want to display only the original, I'll type "original" or only mine, I'll type "plugin" or both, I'll type "both" and if is both, I need to decide the order, if original first or mine first, right?

Reply:
Throwing an existing use case @emmaindal has:

* Say we have set a global support link for the entire instance.
* What if the techdocs plugin now also has a support link that say submits a msg only to a Technical Writing team, instead of a Backstage integrator.
* Do we MERGE the two, that is show a link BOTH for the Backstage integrator team as WELL as the technical writer team? Or does the technical writer team OVERWRITE it?  Or do we allow the team supplying the plugin/specific config allow to identify that (merge vs. overwrite)?

I think that's the thought?

Reply:
thank you so much for adding my use cases here @adamdmharvey I think this should be your 3 idea, that the team responsible for the plugin can decide if they want only the techdocs support information or if they want all support information to show up! 

what if it's also possible in the config to specify which plugin a config belongs to? so that I can pass in a prop to the supportbutton of plugin name/relation or something, like "techdocs" -> renders only the techdocs config. if nothing is passed in, the support button renders all configs specified?

Reply:
Thanks @adamdmharvey & @emmaindal. This is more or less what I was referring. The `app.support` key in the `app-config.yaml` was meant meant as a global backstage support which most likely would be managed by the team maintaining the actual backstage instance. Where any custom support items per instance of the `SupportButton` component would be contextual (e.g. the support for the plugin containing it).

I hadn't really thought of the case where the instance/plugin of the button might want to hide the global list. In that case having a support link somewhere at a higher level then the plugin might be additional needed. We actually added a "Support" side bar nav link that uses the `app.support.url` field which serves this.

So probably the `SupportButton` needs the following attributes/features?

- `title/description` - required field that describes what the support button is referencing. Right now that's all the children, but that probably needs to be more contextual through a well known sub component or a prop instead of all children.
- custom `items/supportItems` - optional additional support items either as a prop or directly children that are slotted into the correct location.
- Ordering of `app.support.items` and custom items - I would think plugin specific should go on top of the global config ones?
- some sort of `mode` or flag that allows each instance to control how the `app.support.items` are used. Maybe this is just `itemsMode="customOnly", itemsMode="all"`?